docs/source/workflow-apis.md

```diff
@@ -0,0 +1,170 @@
+# Workflow APIs
+
+## Using Workflow APIs
+
+To start using a Workflow API you have to know which API you are going to be using.
+In this case we are assuming that the repository defines a v2 namespace with an API definition of MyExampleAPI.
+
+Now to use it, you have to import the API class object you want to use and specify the type of the class in the `apis`
+field.
+This is necessary so all the messages that are consumed and produced by the API can be picked up by the framework
+and the actor can depend on them.
+
+Here is an example of the usage:
+
+```python
+from leapp.actors import Actor
+from leapp.workflows.api.v2 import MyExampleAPI
+
+class MyWorkflowAPIUsingActor(Actor):
+    """ An example actor consuming the v2.MyExampleAPI Workflow API """
+    produces = ()
+    consumes = ()
+    apis = (MyExampleAPI,)
+
+    def process(self):
+        api = MyExampleAPI()
+        api.second(value=10)
+```
+
+## General workings of the Workflow API feature
+
+In the leapp repository , you want to create the API, for create a folder called `apis`.
+Now within the `apis` folder anything that is in there can be imported with the `leapp.workflows.api` prefix.
+An example: 
+
+You created a module `$repository/apis/example.py` this module can now be imported like this:
+
+```python
+from leapp.workflows.api import example
+```
+
+## Defining Workflow APIs
+
+A basic Workflow API is defined like this:
+
+```python
+from leapp.workflows.api import WorkflowAPI
+
+class MyExampleAPI(WorkflowAPI):
+    pass
+```
+
+This of course is no good as this wouldn't do anything just yet. APIs are nothing else than python classes and one
+just adds a method to it to make it available.
+
+Here is a simple function added to the API:
+
+```python
+from leapp.workflows.api import WorkflowAPI
+
+class MyExampleAPI(WorkflowAPI):
+    def first(self):
+        return 'First API function result'
+```
+
+### Working with messages
+
+Now if your API is going to consume or produce any messages, those messages have to be declared the same way as they
+are declared for actors.
```

Comment:
Add a note here (or below, but in the same section) that an actor using an api doesn't have to explicitly name all messages consumed/produced by the API, as leapp framework unifies equivalent tuples. It could be short, with reference on dependencies.